tests/SoldiersAPI.test.ts

```diff
@@ -0,0 +1,42 @@
+import request from 'supertest';
+import router from '../src/routes/routes';
+import * as soldiers from './testData';
+import { getDatabase, connectToMongoDB } from '../src/repository/mongoConnect';
+
+const app = router;
+beforeEach(async () => {
+  await connectToMongoDB();
+  const db = await getDatabase();
+  await db.collection('soldiers').deleteMany({});
+});
+describe('Insert Soldier into DB', () => {
+  it('Should Send an API Post Request and save soldier to DB, Expected 201:', async () => {
+    await request(app).post('/soldiers').send(soldiers.SoldierToBeAddedForPOSTtest).expect(201);
+  });
+});
+
+describe('Fail to insert Soldier into DB', () => {
+  it('Should Send an API Post Request and not save soldier to DB, Expected 400:', async () => {
+    await request(app).post('/soldiers').send(soldiers.SoldierToBeAddedForFailedPOSTtest).expect(400);
+  });
+});
+
+describe('Find Soldier By ID', () => {
+  it('Should Send an API Get Request If Soldier Is In DB, Expected 200:', async () => {
+    request(app).post('/soldiers/fuck').send(soldiers.SoldierToBeFound);
+    request(app).get('/soldiers/1').expect(soldiers.SoldierToBeFound);
+    request(app).get('/soldiers/505').expect(404);
+  });
+});
+
+describe('Search for Soldier by query in DB', () => {
+  it('Should Send an API Get Request and see if Soldier is in DB, Expected 200:', async () => {
+    request(app).get('/soldiers?name=Darth Vader&rank=Silver').expect(soldiers.SoldierToSearch);
```

Comment:
Since you are testing for a soldier that you added a moment ago, make sure that your search is with the correct fields.

```JS
testSoldier = {
  "name": "Darth",
  "rank": "silver",
}
```

Then here you do:
```JS
request(app).get(`/soldiers?name=${testSoldier.name}&rank=${testSoldier.rank}`)
```